rpg.py

```diff
@@ -246,18 +254,22 @@ def level_up(self):
             if player.is_alive():
                 player.exp += exp_gain
                 print(f'{player.name} gained {exp_gain} experience points!')
-                if player.exp >= player.target_exp:
+                if player.exp >= player.level_exp:
                     print(f'{player.name} leveled up!')
                     player.level += 1
                     player.max_hp += 2
                     player.hp = player.max_hp
                     player.attack += 1
                     player.speed += 1
-                    player.target_exp *= 2
+                    if player.leveling_rate == 1:
+                        player.level_exp *= 2
+                    elif player.leveling_rate == 2:
+                        player.level_exp *= 4
```

Comment:
At this point you should use a formula: `player.level_exp *= (player.leveling_rate * 2)`